### DISCUSSION TITLE
Code Coverage for the native code

### DISCUSSION THREAD

**Discussion by malak-abbas** [android/ndk-samples#991] [Q&A] [answered]
Is there a way to generate code coverage report for the native cpp with some gradle options like we do with jaCoCo (for java)?

**Comment by DanAlbert** [comment, marked as answer]
There's some discussion [here](https://github.com/android/ndk/issues/612). You can rig something up if you're content to build the infrastructure yourself, but I wouldn't call it a great experience.

Doing better is really a question for Android Studio. Try filing a feature request for them: http://b.android.com
